lib/cli-options.js

```diff
@@ -12,15 +15,80 @@ exports.CLIOptions = class {
     return parts.join('.');
   }
 
+  getProject() {
+    return JSON.parse(readFileSync([process.cwd(), 'aurelia_project', 'aurelia.json'].join('/')));
+  }
+
   getEnvironment() {
+    const project = this.getProject();
     let NODE_ENV;
-    let env = this.getFlagValue('env') || (process.env.NODE_ENV ? NODE_ENV = process.env.NODE_ENV : undefined) || 'dev';
+    let env = this.getFlagValue('env') || (process.env.NODE_ENV ? NODE_ENV = process.env.NODE_ENV : undefined) || project.build.defaultEnvironment || 'dev';
     if (NODE_ENV) {
       console.log(`The selected Node Environment (${NODE_ENV}) is not a preconfigured option ('dev', 'stage', and 'prod')`);
     }
     return env;
   }
 
+  getTarget(targetId, env) {
+    const project = this.getProject();
+    const targets = project.build.targets;
+    targetId = targetId || project.build.defaultTarget || 'web';
+    env = env || this.getEnvironment();
+
+    let target;
+    if (Array.isArray(targets)) {
+      target = targets[0];
+      target.legacy = true;
```

Comment:
Perhaps I'm missing something, but is this always going to return the first target if the `project.build.targets` is in fact an array?

Do you perhaps have an example of what the structure should look like?
My `aurelia.json` currently looks like this:
```
"build": {
    "targets": [
      {
        "id": "web",
        "displayName": "Web",
        "index": "index.html",
        "baseDir": ".",
        "output": "scripts"
      },
      {
        "id": "kiosk",
        "displayName": "Kiosk",
        "index": "kiosk.html",
        "baseDir": ".",
        "output": "kiosk_scripts"
      }
    ]
```

Reply:
@johan-v-r You're not missing something; that is the current behaviour (which is kept so that it's a non-breaking change).

Here's a (somewhat long) example:
```
    "targets": {
      "web": {
        "id": "web",
        "displayName": "Web",
        "index": "index.html",
        "output": "scripts",
        "root": "dist/${TARGET}/${ENV}",
        "resources": [
          "favicon.ico",
          "images/**",
          "styles/**"
        ],
        "environments": {
          "dev": {
            "root": "build/dev"
          }
        }
      },
      "aspnetcore": {
        "id": "aspnetcore",
        "displayName": "ASP.NET Core",
        "index": "index.html",
        "output": "scripts",
        "root": "wwwroot",
        "resources": [
          "favicon.ico",
          "images/**",
          "styles/**"
        ],
        "environments": {
          "prod": {
            "root": "../dotnetcore/wwwroot"
          }
        }
      },
      "cordova": {
        "id": "cordova",
        "displayName": "Cordova",
        "index": "index.html",
        "indexSource": "index.cordova.html",
        "output": "scripts",
        "root": "../cordova/www",
        "resources": [
          "favicon.ico",
          "images/**",
          "styles/**"
        ]
      },
      "EXTERNAL": "aspnetcore & cordova",
      "ALL": "web & EXTERNAL"
    }
```
In this example the Aurelia project also contains a dotnet core web server/project (not really a good idea, but it's just an example) that will be targeted for dev and stage builds. In addition, the Aurelia project has two "sibling projects", one Cordova and one that's, again, dotnet core (which is targeted for the prod build). The Cordova project has additional requirements for index.html, so it's a separate file in the Aurelia project (that will be renamed to index.html for the target).

Running 
`au build --target ALL --watch`
will rebuild and update all three targets as soon as a file is changed.

Reply:
Okay yeah thanks this is awesome, will 100% work for me. This also replaces my PR #731   

Just a final note from my side, looks like `root` property is more like an output directory..? Can/should this name perhaps be changed?

Reply:
@johan-v-r Well, it is the target root and the base for the other target settings. (And, depending on them, nothing might actually end up there.) Plus, I didn't want it mixed up with the current `output` (which is where the script bundles are put). But yeah, perhaps it should be changed (but then `output` also needs to be changed). Naming things is often the hardest part of this stuff...